test/unit/builtins/map.spec.ts

```diff
@@ -135,3 +135,45 @@ test("map size", () => {
     expect(util.transpileAndExecute(`let m = new Map([[1,2],[3,4]]); m.clear(); return m.size;`)).toBe(0);
     expect(util.transpileAndExecute(`let m = new Map([[1,2],[3,4]]); m.delete(3); return m.size;`)).toBe(1);
 });
+
+const testMapConstructionCode = `
```

Comment:
Have you considered using .each?

```ts
test.each(["keys", "values", "entries"])("map.%s() preserves insertion order", (method) => {
    util.testFunction`
        const mymap = new Map();

        mymap.set("x", 1);
        mymap.set("a", 2);
        mymap.set(4, 3);
        mymap.set("b", 6);
        mymap.set(1, 4);
        mymap.set("a", 5);
        
        mymap.delete("b");

        return [...mymap.${method}()];
    `.expectToMatchJsResult();
});
```

Reply:
That would mean I can no longer also check for the actual expected value. I have not made my mind up on if that should be there or not yet.